### DISCUSSION TITLE
How do I configure multiple NS Panels?

### DISCUSSION THREAD

**Discussion by MarkuZP** [Blackymas/NSPanel_HA_Blueprint#230] [Q&A] [answered]
Hello community!

I have a noob question  here :-)
I am wondering how I set up multiple Panels with ESPHome / Blueprints.
Must I add an ESP Device for Each Panel or only individual Blueprints?

Thnx!

**Comment by tikismoke** [comment, marked as answer]
I will say one of both for each to allow esphome code to be sent to the right device. And automation per nspanel to allow right information/action

**Comment by ghost** [comment]
@tikismoke is correct, you have an esphome config per device for it to function / connect to network, and a home assistant automation per device to integrate it

**Comment by MarkuZP** [comment]
Thank you all! That's what I guessed but I had a strange behaviour with the new ESP Home device. After flashing the name of the Device in the Devices Tab of HA was still the name of the first ESP Home device /NS Panel!?

Weird!
<img width="881" alt="Bildschirm­foto 2023-01-24 um 08 32 19" src="https://user-images.githubusercontent.com/109298227/214235346-b7c95d43-ec65-48d7-9312-8faa46193678.png">
As you can see nspaneleg is online and working but the only device in HA is nspanelblacky which was my first nspanel I flashed.

**Reply by robgough1970** [reply]
in HA go to - settings > Devices and services > add integration > esphome then add the second panels IP address and this should pull the esphome device into HA.
